vendor/github.com/kubernetes-csi/external-snapshotter/client/v3/apis/volumesnapshot/v1/types.go

```diff
@@ -0,0 +1,395 @@
+type VolumeSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// List of VolumeSnapshots
+	Items []VolumeSnapshot `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// VolumeSnapshotSpec describes the common attributes of a volume snapshot.
+type VolumeSnapshotSpec struct {
+	// source specifies where a snapshot will be created from.
+	// This field is immutable after creation.
+	// Required.
+	Source VolumeSnapshotSource `json:"source" protobuf:"bytes,1,opt,name=source"`
+
+	// volumeSnapshotClassName is the name of the VolumeSnapshotClass requested by the VolumeSnapshot.
+	// If not specified, the default snapshot class will be used if one exists.
+	// If not specified, and there is no default snapshot class, dynamic snapshot creation will fail.
+	// Empty string is not allowed for this field.
+	// TODO(xiangqian): a webhook validation on empty string.
+	// More info: https://kubernetes.io/docs/concepts/storage/volume-snapshot-classes
+	// +optional
+	VolumeSnapshotClassName *string `json:"volumeSnapshotClassName,omitempty" protobuf:"bytes,2,opt,name=volumeSnapshotClassName"`
+}
+
+// VolumeSnapshotSource specifies whether the underlying snapshot should be
+// dynamically taken upon creation or if a pre-existing VolumeSnapshotContent
+// object should be used.
+// Exactly one of its members must be set.
+// Members in VolumeSnapshotSource are immutable.
+// TODO(xiangqian): Add a webhook to ensure that VolumeSnapshotSource members
+// will not be updated once specified.
+type VolumeSnapshotSource struct {
+	// persistentVolumeClaimName specifies the name of the PersistentVolumeClaim
+	// object in the same namespace as the VolumeSnapshot object where the
+	// snapshot should be dynamically taken from.
+	// This field is immutable.
+	// +optional
+	PersistentVolumeClaimName *string `json:"persistentVolumeClaimName,omitempty" protobuf:"bytes,1,opt,name=persistentVolumeClaimName"`
+
+	// volumeSnapshotContentName specifies the name of a pre-existing VolumeSnapshotContent
+	// object.
+	// This field is immutable.
+	// +optional
+	VolumeSnapshotContentName *string `json:"volumeSnapshotContentName,omitempty" protobuf:"bytes,2,opt,name=volumeSnapshotContentName"`
+}
+
+// VolumeSnapshotStatus is the status of the VolumeSnapshot
+type VolumeSnapshotStatus struct {
+	// boundVolumeSnapshotContentName represents the name of the VolumeSnapshotContent
+	// object to which the VolumeSnapshot object is bound.
+	// If not specified, it indicates that the VolumeSnapshot object has not been
+	// successfully bound to a VolumeSnapshotContent object yet.
+	// NOTE: Specified boundVolumeSnapshotContentName alone does not mean binding
+	//       is valid. Controllers MUST always verify bidirectional binding between
+	//       VolumeSnapshot and VolumeSnapshotContent to avoid possible security issues.
+	// +optional
```

Comment:
```suggestion
	// boundVolumeSnapshotContentName is the name of the VolumeSnapshotContent
	// object to which this VolumeSnapshot object intends to bind to.
	// If not specified, it indicates that the VolumeSnapshot object has not been
	// successfully bound to a VolumeSnapshotContent object yet.
	// NOTE: To avoid possible security issues, consumers must verify binding between
	// VolumeSnapshot and VolumeSnapshotContent objects is successful (by validating that
	// both VolumeSnapshot and VolumeSnapshotContent point at each other) before using
	// this object.
	// +optional
```

Reply:
Done